### PR TITLE
Fix "The variable ‚rows‘ already exists"

### PR DESCRIPTION
**Pull request by temistokles**

### CODE REVIEW

**Comment by paveljanda** [comment]
Hm, I know. But it is an issue in nette components. :/

@juniwalk Your opinion?

**Comment by juniwalk** [comment]
@paveljanda Well there is not much of a choice. If there already is any variable this is the only way to overwrite it, as far as I know.

**Comment by juniwalk** [comment]
That `$this->getTemplate()` should be refactored into variable. Soooo much calls for no reason what so ever.

File: src/DataGrid.php [hunk 1]
```diff
@@ -511,34 +511,34 @@ public function render()
...
-		$this->getTemplate()->add('hasGroupActionOnRows', $hasGroupActionOnRows);
+		$this->getTemplate()->hasGroupActionOnRows = $hasGroupActionOnRows;
 
 		/**
 		 * Walkaround for Latte (does not know $form in snippet in {form} etc)
 		 */
-		$this->getTemplate()->add('filter', $this['filter']);
-
+		$this->getTemplate()->filter = $this['filter'];
+
```
Comment (paveljanda) [inline src/DataGrid.php]: Can you please remove this empty line identation? Thanks.

Comment (temistokles) [inline src/DataGrid.php]: There you go :)

**Comment by radekdostal** [comment]
Alternative form of setting is
```php
$this->getTemplate()->setParameters([
  'variable1' => $value1,
  'variable2' => $value2,
  ...
]);
```

**Comment by paveljanda** [comment]
Thanks!

**Comment by juniwalk** [comment]
@radekdostal Does it scream with errors if there already is parameter with the same name?

**Comment by radekdostal** [comment]
@juniwalk I think that add() method throws exception when the variable already exists, but setTemplate() allows it to overwrite.

**Pull request merged**